internal/rpc/user_device_data_api.go

```diff
@@ -85,13 +86,54 @@ func (s *userDeviceData) GetUserDeviceData(ctx context.Context, req *pb.UserDevi
 	}, nil
 }
 
+// ? TODO: should this follow mediator patter as all other services?
+func (s *userDeviceData) GetRawDeviceData(ctx context.Context, req *pb.RawDeviceDataRequest) (*pb.RawDeviceDataResponse, error) {
+	if isEmpty(req.UserDeviceId) {
+		return nil, status.Error(codes.InvalidArgument, "UserDeviceId is required")
+	}
+
+	query := make([]qm.QueryMod, 0)
+
+	query = append(query, models.UserDeviceDatumWhere.UserDeviceID.EQ(req.UserDeviceId))
+
+	if req.IntegrationId != nil && !isEmpty(*req.IntegrationId) {
```

Comment:
i would just expect integration_id to always be required

Reply:
actually nevermind, all good

Reply:
this is marked as optional since in some scenarios we don't have integration id at hand.